### ISSUE TITLE
Unexpected error in `FrobeniusCharacterValue(E(16),269)`

### ISSUE THREAD

**Issue by jackschmidt** [gap-system/gap#4564]
There seems to be a problem with `FrobeniusCharacterValue` for some medium sized fields. The first problem I found for E(16) was GF(269).

```
gap> for p in Primes do FrobeniusCharacterValue(E(16),p); od;
```

I didn't get an error for E(4) or E(8). E(32) has an error at p=263.

### Observed behaviour

```
gap> FrobeniusCharacterValue(E(16),269);
Error, AddRowVector: <dst> must have the same length as <src> (lengths are 4 and 5) in
  AddRowVector( vec, sem.vectors[vno], - x
 ); at /home/jack/devel/gap/lib/matrix.gi:2806 called from
SolutionMatDestructive( fieldbase, value
 ) at /home/jack/devel/gap/lib/ctblfuns.gi:4218 called from
<function "FrobeniusCharacterValue">( <arguments> )
 called from read-eval loop at *stdin*:1
type 'quit;' to quit to outer loop
brk>
```

### Expected behaviour

```
gap> FrobeniusCharacterValue(E(16),269);
162+256z+143z2+219z3
```

### Copy and paste GAP banner (to tell us about your setup)

```
$ ./gap -A
 ┌───────┐   GAP 4.12dev-940-g2aa151a built on 2021-06-14 21:42:26-0400
 │  GAP  │   https://www.gap-system.org
 └───────┘   Architecture: x86_64-pc-linux-gnu-default64-kv8
 Configuration:  gmp 6.2.1, GASMAN, readline
 Loading the library and packages ...
 Packages:   GAPDoc 1.6.4, PrimGrp 3.4.1, SmallGrp 1.4.2, TransGrp 3.2
 Try '??help' for help. See also '?copyright', '?cite' and '?authors'
gap> FrobeniusCharacterValue(E(16),269);
Error, AddRowVector: <dst> must have the same length as <src> (lengths are 4 and 5) in
  AddRowVector( vec, sem.vectors[vno], - x
 ); at /home/jack/devel/gap/lib/matrix.gi:2806 called from
SolutionMatDestructive( fieldbase, value
 ) at /home/jack/devel/gap/lib/ctblfuns.gi:4218 called from
<function "FrobeniusCharacterValue">( <arguments> )
 called from read-eval loop at *stdin*:1
type 'quit;' to quit to outer loop
brk>
```

```
$ gap -A
 ┌───────┐   GAP 4.11.1 of 2021-03-02
 │  GAP  │   https://www.gap-system.org
 └───────┘   Architecture: x86_64-pc-linux-gnu-default64-kv7
 Configuration:  gmp 6.2.1, GASMAN, readline
 Loading the library and packages ...
 Packages:   GAPDoc 1.6.4, PrimGrp 3.4.1, SmallGrp 1.4.2, TransGrp 3.0
 Try '??help' for help. See also '?copyright', '?cite' and '?authors'
gap> FrobeniusCharacterValue(E(16),269);
Error, AddRowVector: vector lengths differ <left> 4,  <right> 5 in
  AddRowVector( vec, sem.vectors[vno], - x ); at /usr/lib/gap/lib/matrix.gi:2752 called from
SolutionMatDestructive( fieldbase, value ) at /usr/lib/gap/lib/ctblfuns.gi:4218 called from
<function "FrobeniusCharacterValue">( <arguments> )
 called from read-eval loop at *stdin*:1
you can replace vector <right> via 'return <right>;'
brk>
```

**Comment by fingolfin** [comment]
Perhaps @ThomasBreuer has some insights on this.

**Comment by ThomasBreuer** [comment]
The problem is that some trailing zeros of vectors in `fieldbase` do not get removed before the call to `SolutionMatDestructive`.
(The code deals with the situation that the vectors get too short after some reduction, but not with the situation that the vectors are too long.)
The given input had worked in older GAP versions (I tried 4.4.12), but the code of `FrobeniusCharacterValue` was changed afterwards.
I will create a pull request for fixing the problem.
